src/color-thief.js

```diff
@@ -18,45 +18,52 @@
 
 
 /*
-  CanvasImage Class
-  Class that wraps the html image element and canvas.
-  It also simplifies some of the canvas context manipulation
-  with a set of helper functions.
-*/
-var CanvasImage = function (image) {
-    this.canvas  = document.createElement('canvas');
-    this.context = this.canvas.getContext('2d');
-
-    document.body.appendChild(this.canvas);
-
-    this.width  = this.canvas.width  = image.width;
+ CanvasImage Class
+ Class that wraps the html image element and canvas.
+ It also simplifies some of the canvas context manipulation
+ with a set of helper functions.
+ */
+var isValidFocusRect = function (focusRect) {
+  return !!focusRect && typeof focusRect === 'object' && focusRect.top && focusRect.left && focusRect.width && focusRect.height;
```

Comment:
Unsure if width + height can be 0 here, but it seems top and left should be able to 
```suggestion
  return !!focusRect && typeof focusRect === 'object' && focusRect.top !== undefined && focusRect.left !== undefined && focusRect.width !== undefined && focusRect.height !== undefined;
```